modules/core/src/main/java/org/apache/synapse/mediators/builtin/CallMediator.java

```diff
@@ -131,10 +173,24 @@ public void setEndpoint(Endpoint endpoint) {
 
     public void init(SynapseEnvironment synapseEnvironment) {
         this.synapseEnv = synapseEnvironment;
+
         if (endpoint != null) {
             endpoint.init(synapseEnvironment);
         }
-        synapseEnvironment.updateCallMediatorCount(true);
+        if (blocking == true) {
+            try {
+                configCtx = ConfigurationContextFactory.createConfigurationContextFromFileSystem(
+                        DEFAULT_CLIENT_REPO, DEFAULT_AXIS2_XML);
+                blockingMsgSender = new BlockingMsgSender();
+                blockingMsgSender.setConfigurationContext(configCtx);
+                blockingMsgSender.init();
+
+            } catch (AxisFault axisFault) {
+                axisFault.printStackTrace();
```

Comment:
I should log the message and throw the synapse exception
log.error(msg, axisFault);
throw new SynapseException(msg, axisFault);

the suggested changes are fixed in the latest code